### PR TITLE
feat: Spanner copy backup

### PR DESCRIPTION
**Pull request by asthamohta**
The following functionalities are added in this PR:
1. Support for copyBackup in the Node client
2. Samples for copyBackup
3. Support for maxExpireTime and referencingBackup
4. Updating update backup sample to reflect how to use maxExpireTime

### CODE REVIEW

File: samples/backups-copy.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// sample-metadata:
+//  title: Read data using an existing index.
+//  usage: node spannerCopyBackup <INSTANCE_ID> <DATABASE_ID> <PROJECT_ID>
```
Comment (vi3k6i5) [inline samples/backups-copy.js:17]: nit: the hierarchy is Project > Instance > database, maybe we can update this to follow that, but not necessary.

File: samples/system-test/spanner.test.js [hunk 2]
```diff
@@ -63,6 +64,7 @@ const DEFAULT_LEADER_2 = 'us-east1';
 
 const spanner = new Spanner({
   projectId: PROJECT_ID,
+  apiEndpoint: 'staging-wrenchworks.sandbox.googleapis.com',
```
Comment (vi3k6i5) [inline samples/system-test/spanner.test.js:67]: this i don't think we should commit to main branch.

File: src/instance.ts [hunk 3]
```diff
@@ -249,6 +250,24 @@ class Instance extends common.GrpcServiceObject {
...
 
+  copyBackup(
+    sourceBackupId: string,
+    backupId: string,
+    options: CopyBackupOptions,
+    callback?: CopyBackupCallback
+  ): Promise<CopyBackupResponse> | void {
+    if (!backupId || !sourceBackupId) {
+      throw new Error(
+        'A backup ID and source backup ID is required to create a Backup.'
```
Comment (vi3k6i5) [inline src/instance.ts:261]: nit: *to create a copy of the backup.

File: samples/backups-copy.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  projectId = 'my-project-id'
+) {
+  // [START spanner_copy_backup]
+  /**
+   * TODO(developer): Uncomment these variables before running the sample.
+   */
+  // const instanceId = 'my-instance';
+  // const databaseId = 'my-database';
+  // const backupId = 'my-backup',
+  // sourceBackupId = 'my-source-backup',
```
[the remark this answers is not in the document]
Comment (asthamohta) [reply samples/backups-copy.js:35]: My bad, will make it a const

File: samples/backups-copy.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    projectId: projectId,
+  });
+
+  async function spannerCopyBackup() {
+    // Gets a reference to a Cloud Spanner instance, database and backup
+    const instance = spanner.instance(instanceId);
+    const database = instance.database(databaseId);
+    const sourceBackup = instance.backup(sourceBackupId);
+
+    // Expire source and copy backup 14 days in the future
```
[the remark this answers is not in the document]
Comment (asthamohta) [reply samples/backups-copy.js:53]: My bad, initially I was creating a backup and then copying it, later decided to use an existing one, forgot to change the comment

**Review by vi3k6i5 — APPROVED** [review]
LGTM

**Review by vi3k6i5 — APPROVED** [review]
LGTM except small nit change.

File: samples/README.md [hunk 6]
```diff
@@ -15,6 +15,7 @@ and automatic, synchronous replication for high availability.
 * [Before you begin](#before-you-begin)
 * [Samples](#samples)
   * [Backups-cancel](#backups-cancel)
+  * [Copies a source backup](#copies-a-source-backup)
```
[the remark this answers is not in the document]
Comment (asthamohta) [reply samples/README.md:18]: That is because this is in the new format of test.  As you can see on line 34-36, this is acceptable

**Pull request merged**